File: Lesson9/recyclerviewapp/src/main/java/ru/mirea/gupalodr/recyclerviewapp/MainActivity.java
package ru.mirea.gupalodr.recyclerviewapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
// Cоздать список исторических событий с кратким описанием и изображением.
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<Event> events = getListData();
        RecyclerView recyclerView = this.findViewById(R.id.recyclerView);
        recyclerView.setAdapter(new EventRecyclerViewAdapter(events));

        // RecyclerView scroll vertical
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        recyclerView.setLayoutManager(layoutManager);
    }
    private List<Event> getListData() {
        List<Event> list = new ArrayList<Event>();
        Event cosmos = new Event("Первый полет в космос",
                "12 апреля 1961 года на околоземную орбиту вышел Юрий Алексеевич Гагарин",
                "vostok_spacecraft");
        list.add(cosmos);
        return list;
    }

}

File: Lesson3/IntentApp/src/main/java/ru/mirea/gupalodr/lesson3/MainActivity.kt
package ru.mirea.gupalodr.lesson3

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import java.text.SimpleDateFormat
import java.util.Date

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<Button>(R.id.button).setOnClickListener {
            val intent: Intent = Intent(this, SecondActivity::class.java)

            val dateInMillis = System.currentTimeMillis()
            val format = "yyyy-MM-dd HH:mm:ss"
            val sdf = SimpleDateFormat(format)
            val dateString: String = sdf.format(Date(dateInMillis))

            intent.putExtra("currentTime", dateString)
            startActivity(intent)
        }
    }
}